Makefile

```diff
@@ -15,7 +15,7 @@
 
 CFLAGS ?= -O2 -Wall -Wshadow
 
-GIT_VERSION := $(shell git describe --abbrev=6 --dirty --always || date +%F)
+GIT_VERSION := $(shell type git >/dev/null && git describe --abbrev=6 --dirty --always || date +%F)
```

Comment:
This assumes SHELL=*sh, and if this is so, I'd like to use type -P instead.

However, I tested this by git->nogit to trigger the missing binary, and it just works:

```
[nut:~/working-repos/carbon-c-relay] % make CC=clang
/bin/sh: nogit: command not found
clang -O2 -Wall -Wshadow -DGIT_VERSION=\"2016-09-21\" -pthread   -c -o relay.o relay.c
...
```

so I'd like to understand why it fails for you